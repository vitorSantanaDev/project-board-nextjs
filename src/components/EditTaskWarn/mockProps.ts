export const handleClickCancelEdit = jest.fn()
