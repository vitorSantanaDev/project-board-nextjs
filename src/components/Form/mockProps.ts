export const formProps = {
  handleAddTaskSubmit: jest.fn(),
  handleChangeInput: jest.fn(),
  input: ''
}
