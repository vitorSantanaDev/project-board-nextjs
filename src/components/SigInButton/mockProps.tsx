export const mockProps = {
  session: true,
  onClick: jest.fn()
}
