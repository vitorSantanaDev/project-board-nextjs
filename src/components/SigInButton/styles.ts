import styled from 'styled-components'

export const ButtonWrapper = styled.button`
  height: 2.5rem;
  border-radius: 3rem;
  cursor: pointer;

  padding: 0 1.5rem;
  margin-left: auto;

  color: #fff;
  background-color: transparent;

  font-weight: bold;

  display: flex;
  align-items: center;
  justify-content: center;

  transition: ease-in-out 0.3s;

  img {
    width: 45px;
    height: 45px;
    border-radius: 50%;
    margin-right: 1.2rem;
  }

  svg {
    width: 30px;
    height: 30px;

    &:first-child {
      margin-right: 1rem;
    }
  }

  svg.logoutIcon {
    margin-left: 1.2rem;
  }

  &:hover {
    opacity: 0.8;
  }
`
