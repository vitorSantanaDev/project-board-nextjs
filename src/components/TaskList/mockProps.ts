export const tasksProps = [
  {
    id: '3896433',
    userId: '8458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '39033',
    userId: '8456458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '38933',
    userId: '865658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '3366553',
    userId: '85654658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '337763',
    userId: '88',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  }
]
