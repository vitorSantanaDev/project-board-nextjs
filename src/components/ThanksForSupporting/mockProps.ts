export const mockProps = {
  lastDonateUser: `Sua última doação foi a ${new Date().toLocaleDateString()}`
}
