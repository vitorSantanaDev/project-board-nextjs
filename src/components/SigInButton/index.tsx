import { signIn, signOut, useSession } from 'next-auth/react'
import { FaGithub } from 'react-icons/fa'
import { FiLogOut } from 'react-icons/fi'
import * as S from './styles'

const SigInButton: React.FC = () => {
  const session = useSession()
  return (
    <>
      {session ? (
        <S.ButtonWrapper type="button" onClick={() => signOut()}>
          <img src="/images/person-exemple.png" alt="Photo do usário logado" />
          Olá Harry
          <FiLogOut color="#737380" className="logoutIcon" />
        </S.ButtonWrapper>
      ) : (
        <S.ButtonWrapper type="button" onClick={() => signIn('github')}>
          <FaGithub color="#ffb800" />
          Entrar com Github
        </S.ButtonWrapper>
      )}
    </>
  )
}

export default SigInButton
