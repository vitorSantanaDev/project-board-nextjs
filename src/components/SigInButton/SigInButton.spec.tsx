import { render, screen } from '@testing-library/react'
import SigInButton from '.'
import { mockProps } from './mockProps'

describe('<SigInButton />', () => {
  it('should render the username', () => {
    render(
      <SigInButton onClick={mockProps.onClick} session={mockProps.session} />
    )
    const buttonSessionTruthy = screen.getByRole('button', {
      name: /olÃ¡ harry/i
    })
    expect(buttonSessionTruthy).toBeInTheDocument
  })

  it('should render the text button', () => {
    render(
      <SigInButton onClick={mockProps.onClick} session={!mockProps.session} />
    )
    const buttonSessionFalse = screen.getByRole('button', {
      name: /entrar com github/i
    })
    expect(buttonSessionFalse).toBeInTheDocument
  })

  it('should match snapshot', () => {
    const { container } = render(<SigInButton {...mockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
