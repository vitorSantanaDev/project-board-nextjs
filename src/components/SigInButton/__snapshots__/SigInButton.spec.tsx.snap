// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SigInButton /> should match snapshot 1`] = `
.c0 {
  height: 2.5rem;
  border-radius: 3rem;
  cursor: pointer;
  padding: 0 1.5rem;
  margin-left: auto;
  color: #fff;
  background-color: transparent;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
}

.c0 img {
  width: 45px;
  height: 45px;
  margin-right: 1.2rem;
}

.c0 svg {
  width: 30px;
  height: 30px;
}

.c0 svg:first-child {
  margin-right: 1rem;
}

.c0 svg.logoutIcon {
  margin-left: 1.2rem;
}

.c0:hover {
  opacity: 0.8;
}

<button
  class="c0"
  type="button"
>
  <img
    alt="Photo do usário logado"
    src="/images/person-exemple.png"
  />
  Olá Harry
  <svg
    class="logoutIcon"
    color="#737380"
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    style="color: rgb(115, 115, 128);"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
    />
    <polyline
      points="16 17 21 12 16 7"
    />
    <line
      x1="21"
      x2="9"
      y1="12"
      y2="12"
    />
  </svg>
</button>
`;
